### PR TITLE
[WIP] fix #34759 : check whether persistent cart is enabled in Data Helper

### LINKED ISSUE #34759: Persistent Cart is not fully disabled.
### Preconditions (*)

1. Bug exists in 2.0.0 though 2.4-develop
2. Reproducible on all page requests when Persistent Cart is disabled through configuration.

### Steps to reproduce (*)

1. Disable Persistent Cart in Admin.
2. Monitor for calls to Magento\Persistent\Observer\CheckExpirePersistentQuoteObserver ( I used Tideways for this).

### Expected result (*)

1. Expected result: The database should not be called as this feature is disabled.

### Actual result (*)

1. The codebase continues to make a significant number of calls (>20) to the database for persistence even though the feature is disabled.
image

---
Please provide [Severity](https://devdocs.magento.com/guides/vpersistent_cart4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by cmtickle**
### Description (*)

Resolves issue with disabling persistent cart highlighted in #34759 

### Related Pull Requests

(2.4) https://github.com/magento/magento2/pull/34761

### Fixed Issues (if relevant)

1. Fixes magento/magento2#34759 

### Manual testing scenarios (*)

1. Apply this fix
2. Disable Persistent Cart via admin.
3. Monitor for database calls triggered by Magento\Persistent\Observer\CheckExpirePersistentQuoteObserver
4. The database should not be called.

### Questions or comments

**Before Optimisation**
image

**After Optimisation**
image

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Persistent/Helper/Data.php [hunk 1]
```diff
@@ -49,18 +49,26 @@ class Data extends \Magento\Framework\App\Helper\AbstractHelper
      */
     protected $_modulesReader;
 
+    /**
+     * @var \Magento\Store\Model\StoreManagerInterface
+     */
+    protected $storeManager;
```
Comment (andrewbess) [inline app/code/Magento/Persistent/Helper/Data.php]: I think it is bad idea to use `protected` for new properties.
Let's use `private` here.

```suggestion
    private $storeManager;
```

File: app/code/Magento/Persistent/Helper/Data.php [hunk 2]
```diff
@@ -49,18 +49,26 @@ class Data extends \Magento\Framework\App\Helper\AbstractHelper
...
+
     /**
      * @param \Magento\Framework\App\Helper\Context $context
      * @param \Magento\Framework\Module\Dir\Reader $modulesReader
      * @param \Magento\Framework\Escaper $escaper
+     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
```
Comment (andrewbess) [inline app/code/Magento/Persistent/Helper/Data.php]: I think we need to use new parameter as optional

```suggestion
     * @param \Magento\Store\Model\StoreManagerInterface|null $storeManager
```

File: app/code/Magento/Persistent/Helper/Data.php [hunk 3]
```diff
@@ -49,18 +49,26 @@ class Data extends \Magento\Framework\App\Helper\AbstractHelper
...
      */
     public function __construct(
         \Magento\Framework\App\Helper\Context $context,
         \Magento\Framework\Module\Dir\Reader $modulesReader,
-        \Magento\Framework\Escaper $escaper
+	\Magento\Framework\Escaper $escaper,
+	\Magento\Store\Model\StoreManagerInterface $storeManager
```
Comment (andrewbess) [inline app/code/Magento/Persistent/Helper/Data.php]: Please see my comments above and https://github.com/magento/magento2/pull/34761#discussion_r762552255
```suggestion
	\Magento\Framework\Escaper $escaper,
	\Magento\Store\Model\StoreManagerInterface $storeManager = null
```

File: app/code/Magento/Persistent/Helper/Data.php [hunk 4]
```diff
@@ -49,18 +49,26 @@ class Data extends \Magento\Framework\App\Helper\AbstractHelper
...
     ) {
         $this->_modulesReader = $modulesReader;
-        $this->_escaper = $escaper;
+	$this->_escaper = $escaper;
+	$this->storeManager = $storeManager;
```
Comment (andrewbess) [inline app/code/Magento/Persistent/Helper/Data.php]: Please see my comments above and #34761 (comment)

```suggestion
        $this->_escaper = $escaper;
        $this->storeManager = $storeManager ?? \Magento\Framework\App\ObjectManager::getInstance()
                ->get(\Magento\Store\Model\StoreManagerInterface::class);
```

**Review by andrewbess — CHANGES_REQUESTED** [review]
Hello @cmtickle 
Thank you for your contribution
Please check my comments
Thank you in advance

**Comment by andrewbess** [comment]
@magento run all tests

**Comment by cmtickle** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by cmtickle** [comment]
I'll progress this after #34761 to avoid further duplication of effort.

**Pull request closed without merging**